packages/expo/cli/start/server/webpack/ssl.ts

```diff
@@ -0,0 +1,47 @@
+import { certificateFor } from '@expo/devcert';
+import fs from 'fs/promises';
+import * as path from 'path';
+
+import * as Log from '../../../log';
+import { ensureDirectoryAsync } from '../../../utils/dir';
+import { ensureDotExpoProjectDirectoryInitialized } from '../../project/dotExpo';
+
+// TODO: Move to doctor as a prereq.
+
+/** Ensure SSL is setup and environment variables are set. */
+export async function ensureEnvironmentSupportsSSLAsync(projectRoot: string) {
+  if (!process.env.SSL_CRT_FILE || !process.env.SSL_KEY_FILE) {
+    const ssl = await getSSLCertAsync(projectRoot);
+    if (ssl) {
+      process.env.SSL_CRT_FILE = ssl.certPath;
+      process.env.SSL_KEY_FILE = ssl.keyPath;
+    }
+  }
+}
+
+/** Create SSL and write to files in the temporary directory. Exposed for testing. */
+export async function getSSLCertAsync(
+  projectRoot: string
+): Promise<{ keyPath: string; certPath: string } | false> {
+  Log.log(`Creating SSL certificate for localhost (you might need to re-run with sudo)`);
```

Comment:
I don't think we want to encourage running with sudo because everything the process writes will be owned by root. If sudo is required, we want to use it for the precise task that needs it instead of the whole process.